pages/index.js

```diff
@@ -1,7 +1,29 @@
 import Head from "next/head";
 import Link from "next/link";
+import { useState, useEffect } from "react";
 
 export default function About() {
+  const [apiToken, setApiToken] = useState("");
+  const [hasToken, setHasToken] = useState(false);
+
+  useEffect(() => {
+    const token = localStorage.getItem("replicateApiToken");
+    if (token) {
+      setApiToken(token);
+      setHasToken(true);
+    }
+  }, []);
+
+  const handleTokenSubmit = (e) => {
+    e.preventDefault();
+    const token = e.target.token.value.trim();
```

Comment:
I think the more idiomatic way of doing this is via the `FormData` API.

```tsx
const formData = new FormData(e.target)
const token = formData.get("token")
```

Reply:
fixed